src/hooks/use-aggrid-row-filter.ts

```diff
@@ -36,46 +29,57 @@ export type UseAggridRowFilterOutputType = {
 };
 
 const removeElementFromArrayWithFieldValue = (
-    filtersArrayToRemoveFieldValueFrom: FilterType[],
+    filtersArrayToRemoveFieldValueFrom: FilterSelectorType[],
     field: string
 ) => {
     return filtersArrayToRemoveFieldValueFrom.filter(
-        (f: FilterType) => f.field !== field
+        (f: FilterSelectorType) => f.column !== field
     );
 };
 
 const changeValueFromArrayWithFieldValue = (
-    filtersArrayToModify: FilterType[],
+    filtersArrayToModify: FilterSelectorType[],
     field: string,
-    newData: FilterDataType
+    newData: FilterSelectorType
 ) => {
     const filterIndex = filtersArrayToModify.findIndex(
-        (f: FilterType) => f.field === field
+        (f: FilterSelectorType) => f.column === field
     );
     if (filterIndex === -1) {
-        return [
-            ...filtersArrayToModify,
-            {
-                field,
-                data: newData,
-            },
-        ];
+        return [...filtersArrayToModify, newData];
     } else {
         const updatedArray = [...filtersArrayToModify];
-        updatedArray[filterIndex].data = newData;
+        updatedArray[filterIndex] = newData;
         return updatedArray;
     }
 };
 
 export const useAggridRowFilter = (
-    filterSelectorKeys: Record<string, string>,
+    filterStoreParam: FilterStorePropsType,
     updateFilterCallback?: () => void
 ): UseAggridRowFilterOutputType => {
-    const [filters, setFilters] = useState<FilterType[]>([]);
+    const dispatch = useDispatch();
+    const { filterType, filterTab, filterStoreAction } = filterStoreParam;
+    const storeFilter = useSelector(
+        (state: any) => state[filterType][filterTab]
+    );
+    const [filters, setFilters] = useState<FilterSelectorType[]>([]);
```

Comment:
I made this component hybrid, meaning if there are filters in the store, it uses them; otherwise, it retains the current behavior, which is to use the state.